physioqc/metrics/cardiac.py

```diff
@@ -1,8 +1,564 @@
+        )
+        filteredsegment -= np.mean(filteredsegment)
+        if i < numsegs - 1:
+            cardfilteredderivs[
+                i * stepsamples : (i * stepsamples) + segsamples
+            ] += filteredsegment
+            cardfilteredweights[i * stepsamples : (i * stepsamples) + segsamples] += 1.0
+        else:
+            cardfilteredderivs[-segsamples:] += filteredsegment
+            cardfilteredweights[-segsamples:] += 1.0
+    cardfilteredderivs /= cardfilteredweights
+    cardfilteredderivs /= np.std(cardfilteredderivs)
+    if debug:
+        print(peakfreqs)
+        plt.plot(rmsnormderiv)
+        plt.plot(cardfilteredderivs)
+        plt.title("Cardiac signal with peaks")
+        plt.show()
+
+    # C. heartbeats segmentation
+    # The fastest credible cardiac rate is 200 bpm -> 0.3 seconds/heartbeat, so set the dist to be 75%
+    # of that in points
+    thedist = int(0.75 * (0.3 * Fs))
+    peakinfo = operations.peakfind_physio(cardfilteredderivs, thresh=0.1, dist=thedist)
+    if debug:
+        ax = operations.plot_physio(peakinfo)
+        plt.show()
+    thetroughs = peakinfo.troughs
+
+    # D. Similarity Analysis and Exclusion of Unreliable heartbeats
+    numheartbeats = len(thetroughs) - 1
+    scaledpeaklength = 100
+    thescaledheartbeats = np.zeros((numheartbeats, scaledpeaklength), dtype=np.float64)
+    theheartbeatlocs = np.zeros((numheartbeats), dtype=np.float64)
+    theheartbeatcorrs = np.zeros((numheartbeats), dtype=np.float64)
+    heartbeatlist = []
+    for thisheartbeat in range(numheartbeats):
+        heartbeatinfo = {}
+        startpt = thetroughs[thisheartbeat]
+        endpt = thetroughs[thisheartbeat + 1]
+        theheartbeatlocs[thisheartbeat] = (endpt + startpt) / (Fs * 2.0)
+        heartbeatinfo["starttime"] = startpt / Fs
+        heartbeatinfo["endtime"] = endpt / Fs
+        heartbeatinfo["centertime"] = theheartbeatlocs[thisheartbeat]
+        thescaledheartbeats[thisheartbeat, :] = resample(
+            cardfilteredderivs[startpt:endpt], scaledpeaklength
+        )
+        thescaledheartbeats[thisheartbeat, :] -= np.min(
+            thescaledheartbeats[thisheartbeat, :]
+        )
+        thescaledheartbeats[thisheartbeat, :] /= np.max(
+            thescaledheartbeats[thisheartbeat, :]
+        )
+        heartbeatlist.append(heartbeatinfo)
+        if debug:
+            plt.plot(thescaledheartbeats[thisheartbeat, :], lw=0.1, color="#888888")
+    averageheartbeat = np.mean(thescaledheartbeats, axis=0)
+    if debug:
+        plt.plot(averageheartbeat, lw=2, color="black")
+        plt.show()
+    for thisheartbeat in range(numheartbeats):
+        theheartbeatcorrs[thisheartbeat] = pearsonr(
+            averageheartbeat, thescaledheartbeats[thisheartbeat, :]
+        ).statistic
+        heartbeatlist[thisheartbeat]["correlation"] = theheartbeatcorrs[thisheartbeat]
+
+    return heartbeatlist
+
+
+def plotheartbeatqualities(heartbeatlist, totaltime=None):
```

Comment:
I'm wondering if all the plotting functions (or just lines related to plotting in general) should not be integrated in the [visualizations script](https://github.com/physiopy/physioqc/blob/master/physioqc/interfaces/visualizations.py) ?